example/ios/Podfile

```
@@ -1,20 +1,33 @@
-$root_path = "../.."
-$node_modules = "#{$root_path}/node_modules"
-$react_native_path = "#{$node_modules}/react-native"
+# Add the following line at the top of Podfile.
+# Codegen produces files/classes that share names, and it will show the warning.
+# deterministic_uuids option surpresses the warning.
+install! 'cocoapods', :deterministic_uuids => false
 
-require_relative "#{$node_modules}/react-native/scripts/react_native_pods"
-require_relative "#{$node_modules}/@react-native-community/cli-platform-ios/native_modules"
+require_relative '../node_modules/react-native/scripts/react_native_pods'
+require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
 
-platform :ios, "10.0"
+platform :ios, '11.0'
 
-target "SafeAreaViewExample" do
-  use_react_native!(path: $react_native_path)
+ENV['USE_CODEGEN_DISCOVERY'] = '1'
 
-  pod "react-native-safe-area-context", :path => $root_path
-  pod "RNCAsyncStorage", :path => "#{$node_modules}/@react-native-community/async-storage"
-  pod "RNCMaskedView", :path => "#{$node_modules}/@react-native-community/masked-view"
-  pod "RNGestureHandler", :path => "#{$node_modules}/react-native-gesture-handler"
-  pod "RNReanimated", :path => "#{$node_modules}/react-native-reanimated"
-  pod "RNScreens", :path => "#{$node_modules}/react-native-screens"
-end
+target 'example' do
+  config = use_native_modules!
+
+  use_react_native!(
+    :path => config[:reactNativePath],
+    :hermes_enabled => true,
+    :fabric_enabled => true,
+    :app_path => "#{__dir__}/..",
+    :config_file_dir => "#{__dir__}/../../..",
```

```suggestion
    :config_file_dir => "#{__dir__}/../..",
```
If `__dir__` equal to PWD command, it will return `system/user/path/react-native-xxxx/example/ios` and then we should go back to times to root folder, which is  `react-native-xxxx` 

Is is correct ?

---

Ahh I see your point right now

```
    const codegenConfigFileDir = path.join(confifDir, dependency);
 ```
 So the `fileDir` will be outside the library and then will point to that dependency. But the problem with that solution is that I have the other rn libraries in this dir and the codegen scans those dir, which are outside the project

---

Planning to make some  improvements to config_file_dir. We want to be able to pass a direct path to a library, as well as supporting multiple values with an array.

---

Interesting, that seems like a better workaround than the one I have now 😅